src/oop/static_modifier/hard/q3/MainTest.java

```diff
@@ -0,0 +1,152 @@
+        Object[][] constructorArgs = {
+                {value1, String.class}
+        };
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName), String.format(wrongValueMessage, className, attributeName));
+    }
+
+    @Test
+    @Order(5)
+    public void awardIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(methodName, new Class<?>[]{int.class}), String.format(incorrectMethodDefinition, methodName, className));
+        assertTrue(testClass.hasModifier(methodName, new Class<?>[]{int.class}, "public"), String.format(incorrectModifierMessage, methodName, className));
+        assertTrue(testClass.hasReturnType(methodName, new Class<?>[]{int.class}, Void.TYPE), String.format(incorrectReturnType, methodName, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependent1")
+    @Order(6)
+    public void candyStoreClassHasCorrectAwardMethod(String value1, int value2) throws Throwable {
+        Object[][] constructorArgs = {
+                {value1, String.class}
+        };
+        Object classInstance = testClass.createInstance(constructorArgs);
+        Clause[] testSentence = new Clause[]{
+                new StringLiteral("There are " + (initialGummies - value2) + " more to give away after the " + value1 + " award.")
+        };
+        Object[][] arguments = {
+                {value2, int.class}
+        };
+        testClass.callMethod(methodName, arguments, classInstance, testSentence);
+        testClass.setFieldValue(null, initialGummies, staticAttributeName);
+    }
+
+    @Test
+    @Tag("dependent1")
+    @Tag("dependent2")
+    @Tag("dependent3")
+    @Order(7)
+    public void correctMainMethod() throws Throwable {
+        Clause[] clauses = {
+                new StringLiteral("There are 80 more to give away after the Corner Jack award."),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("There are 70 more to give away after the Best Deals in Kelowna award."),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("There are 55 more to give away after the The Apple Mall award."),
+                new Optional(new StringLiteral(" "))
+        };
+        Object classInstance = classObject.createInstance();
```

Comment:
You are calling a static method, so you should not actually need to create an instance of the class.

Reply:
I don't know how else to test it with